app/src/main/kotlin/no/nav/tiltakspenger/vedtak/repository/overgangsstønad/OvergangsstønadVedtakDAO.kt

```diff
@@ -0,0 +1,93 @@
+package no.nav.tiltakspenger.vedtak.repository.overgangsstønad
+
+import kotliquery.Row
+import kotliquery.TransactionalSession
+import kotliquery.queryOf
+import no.nav.tiltakspenger.felles.InnsendingId
+import no.nav.tiltakspenger.felles.OvergangsstønadVedtakId
+import no.nav.tiltakspenger.vedtak.OvergangsstønadPeriode
+import no.nav.tiltakspenger.vedtak.OvergangsstønadVedtak
+import org.intellij.lang.annotations.Language
+import java.time.LocalDateTime
+
+class OvergangsstønadVedtakDAO() {
+    fun hentForInnsending(innsendingId: InnsendingId, txSession: TransactionalSession): List<OvergangsstønadVedtak> {
+        return txSession.run(
+            queryOf(hentOvergangsstønadVedtak, innsendingId.toString())
+                .map { row -> row.toOvergangsstønadVedtak() }
+                .asList,
+        )
+    }
+
+    fun lagre(
+        innsendingId: InnsendingId,
+        overgangsstønadVedtak: List<OvergangsstønadVedtak>,
+        txSession: TransactionalSession,
+    ) {
+        slettOvergangsstønadVedtak(innsendingId, txSession)
+        if (overgangsstønadVedtak == null || overgangsstønadVedtak.isEmpty()) return
+        overgangsstønadVedtak.forEach {
+            lagreOvergangsstønadVedtak(innsendingId, it, txSession)
+        }
+    }
+
+    private fun lagreOvergangsstønadVedtak(
+        innsendingId: InnsendingId,
+        overgangsstønadVedtak: OvergangsstønadVedtak,
+        txSession: TransactionalSession,
+    ) {
+        val id = OvergangsstønadVedtakId.random()
+        txSession.run(
+            queryOf(
+                lagreOvergangsstønadVedtak,
+                mapOf(
+                    "id" to id.toString(),
+                    "innsendingId" to innsendingId.toString(),
+                    "fomDato" to overgangsstønadVedtak.periode.fomDato,
+                    "tomDato" to overgangsstønadVedtak.periode.tomDato,
+                    "innhentet" to overgangsstønadVedtak.innhentet,
+                    "tidsstempelHosOss" to LocalDateTime.now(),
+                ),
+            ).asUpdate,
+        )
+    }
+
+    private fun slettOvergangsstønadVedtak(innsendingId: InnsendingId, txSession: TransactionalSession) {
+        txSession.run(queryOf(slettOvergangsstønadVedtak, innsendingId.toString()).asUpdate)
+    }
+
+    private fun Row.toOvergangsstønadVedtak(): OvergangsstønadVedtak {
+        return OvergangsstønadVedtak(
+            id = OvergangsstønadVedtakId.fromDb(string("id")),
+            periode = OvergangsstønadPeriode(
+                fomDato = string("fomDato"),
+                tomDato = string("tomDato"),
+                datakilde = string("datakilde")
+            ),
+            innhentet = localDateTime("innhentet"),
+        )
+    }
+
+    @Language("SQL")
+    private val lagreOvergangsstønadVedtak = """
+        insert into overgangsstønad_vedtak (
+            id,
+            fomDato,
```

Comment:
Navngivinings-konvensjonen vi har brukt på database-felter er snakecase, altså fom_dato og tom_dato. Så er det *ekstremt* forvirrende å holde styr på når du da skal bruke fom_dato og når du skal bruke fomDato (i values-delen av inserten skal det jo stå :fomDato iom at du i mapOf på linje 46 brukte fomDato), så jeg har begynt å bruke snakecase over hele linja her. Så jeg ville skrevet :fom_dato og mapOf( "fom_dato" to overgangsstønadVedtak.periode.fomDato) osv. Det siste har jeg ikke forankra med noen, men jeg gikk lei av å bomme på navngivinga gang etter gang..